src/scenes/MembershipMintPage/PartnerMembershipMintPage.tsx

```diff
@@ -0,0 +1,178 @@
+        acc[asset.asset_contract.address] = true;
+      }
+
+      return acc;
+    },
+    {}
+  );
+  return Object.keys(partnerNftAddressesOwned);
+}
+
+function generateMerkleProof(encodedAbi1: string, encodedAbi2: string) {
+  const elements = [encodedAbi1, encodedAbi2];
+  const merkleTree = new MerkleTree(elements);
+  return merkleTree.getHexProof(elements[0]);
+}
+
+// Checks to see if the address owns any partner NFTs by
+// retrieving all NFTs owned by a user from opensea,
+// and filtering the list by whitelisted partner contract addresses
+async function detectOwnedPartnerNftsFromOpensea(
+  account: string,
+  addresses: string[]
+): Promise<string[]> {
+  const params = getContractAddressQueryParams(addresses);
+  const response = await fetch(
+    `https://testnets-api.opensea.io/api/v1/assets?owner=${account}${params}`,
+    {}
+  );
+
+  const responseBody = await response.json();
+  // const responseBody = { assets: [] };
+  if (responseBody.assets.length > 0) {
+    return reduceOpenseaAssetsToContractAddresses(responseBody.assets);
+  }
+
+  return [];
+}
+
+async function detectOwnedGeneralCardsFromOpensea(account: string) {
+  const response = await fetch(
+    `https://testnets-api.opensea.io/api/v1/assets?owner=${account}&asset_contract_addresses=${GENERAL_MEMBERSHIP_CONRTACT_ADDRESS}`,
+    {}
+  );
+
+  const responseBody = await response.json();
+  return responseBody.assets.length > 0;
+}
+
+// Turns list of contract addresses into url parameters for the GET call to Opensea
+function getContractAddressQueryParams(addresses: string[]) {
+  return addresses.map((address) => `&asset_contract_addresses=${address}`).join('');
+}
+
+function PartnerMembershipMintPageContent() {
+  const { account } = useWeb3React<Web3Provider>();
+  const contract = useGeneralMembershipCardContract();
+
+  const { getSupply } = useMembershipMintPageActions();
+
+  const [ownedPartnerNfts, setOwnedPartnerNfts] = useState<string[]>([]);
+  const [ownsGeneralCard, setOwnsGeneralCard] = useState(false);
+  const canMintToken = useMemo(
+    () => !ownsGeneralCard && ownedPartnerNfts.length > 0,
+    [ownedPartnerNfts.length, ownsGeneralCard]
+  );
+
+  // Regardless of how many partner NFTs they own, we just need one contract address to mint with, so pick first one in list.
+  const partnerContractAddress = useMemo(() => ownedPartnerNfts[0], [ownedPartnerNfts]);
+
+  // use a ref to track if we called OS, to limit to only calling them once
+  const calledOpenseaRef = useRef(false);
+  useEffect(() => {
+    async function checkIfUserCanMint(account: string, addresses: string[]) {
+      const ownsGeneralCard = await detectOwnedGeneralCardsFromOpensea(account);
+
+      if (ownsGeneralCard) {
+        calledOpenseaRef.current = true;
+        setOwnsGeneralCard(true);
+        return;
+      }
+
+      const ownedPartnerNfts = await detectOwnedPartnerNftsFromOpensea(account, addresses);
+      calledOpenseaRef.current = true;
```

Comment:
Looks like we're setting this to true both here and in the if statement above. Maybe we just call it once above the if statement? Since it's always getting called

Reply:
the idea here was to set `calledOpenseaRef.current = true;` once we were finished successfully calling opensea for each case. 

So in the case where `ownsGeneralCard` is true, we don't need to call OS anymore so we set the ref to true immediately in the if block.

Otherwise, we need to call `detectOwnedPartnerNftsFromOpensea` so we wait until that's successfully called, then set the ref to true after.

To try to make this clearer I've renamed the ref to `finishedCallingOpensea`. Lmk if that makes sense